src/ui/map.js

```diff
@@ -1339,6 +1339,34 @@ class Map extends Camera {
         return this.style.getLight();
     }
 
+    /**
+     * Sets state values on a feature, given a source and [optional]source-layer.
+     * 
+     * @param {string} source The source identifier for the feature.
+     * @param {string} feature The unique feature identifier from the source data.
+     * @param {string} key The name of the state to be set.
+     * @param {string | number | boolean} value The value to be set.
+     * @param {string} sourceLayer (optional) The source-layer identifier for Vector Tile sources.
+     */
+    setFeatureState(source: string, feature: string, key: string, value: any, sourceLayer?: string) {
```

Comment:
This seems like an improvement. I'd like to make the API error out if a source layer is not provided for Vector sources. It would mean checking the type of the source id, but it would prevent erroneous use of the API.